google/cloud/testing_util/assert_ok.h

```diff
@@ -39,6 +39,13 @@ testing::AssertionResult IsOkPredFormat(
 }  // namespace internal
 }  // namespace testing
 
+// This is a dirty workaround for the fact that protobuf defines a poorman's
+// version of this macro for grpc::Status.
+
+#ifdef EXPECT_OK
+#undef EXPECT_OK
+#endif
```

Comment:
Hmm... I was a little worried about this macro conflicting with something. I wonder if we could instead implement this more like a gmock matcher. It would be used like:

```cc
EXPECT_THAT(x, IsOK());
```

IsOk(), would not be a macro, it would be a namespaced C++ class (or function), so it would not conflict with anything? Maybe something like this: https://github.com/google/googletest/blob/master/googlemock/docs/CookBook.md#writing-new-polymorphic-matchers

Reply:
+1

Reply:
@dopiera suggested that the usage with an EXPECT_THAT matcher may be more cumbersome for callers to use throughout tests. It's more syntax to type and would require a using declaration in each file that uses it.

The suggestion was made to just rename this macro, and knowing that macros are horrible, hope that we don't get a build break from a name conflict. That sounds reasonable to me. A future refactoring (if we ever need it) should be a pretty simple search-n-replace.

@devbww and I chatted offline and like the name `EXPECT_STATUS_OK())` (and the assert variant). So, if that works and all the builds pass, just renaming the macro SGTM.